include/boost/circular_buffer/base.hpp

```diff
@@ -1449,6 +1449,42 @@ class circular_buffer
         BOOST_CATCH_END
     }
 
+    template <class ...Args>
```

Comment:
You shouldn't define these function templates unconditionally, but instead, only when you detect that `BOOST_NO_CXX11_VARIADIC_TEMPLATES` is _not_ defined. This library still supports C++ standards prior to C++11.